packages/fs-detectors/test/unit.local-file-system-detector.test.ts

```diff
@@ -0,0 +1,88 @@
+import os from 'node:os';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { LocalFileSystemDetector, DetectorFilesystem } from '../src';
+
+const tmpdir = path.join(os.tmpdir(), 'local-file-system-test');
+
+const dirs = ['', 'a', 'a/b']; // root, single-nested, double-nested
+const files = ['foo', 'bar'];
+const filePaths = dirs.flatMap(dir => files.map(file => path.join(dir, file)));
```

Comment:
This file structure should cover all the relevant bases. root files, single nested files, and multie nested files.